From 151c3ee9d07e2ecf0fbd42090deefa46866f4319 Mon Sep 17 00:00:00 2001
From: Gilles Boccon-Gibod <bok@bok.net>
Date: Sun, 20 Mar 2022 15:24:05 -0700
Subject: [PATCH 12/13] fix #663

---
 Source/C++/Core/Ap4HdlrAtom.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Source/C++/Core/Ap4HdlrAtom.cpp b/Source/C++/Core/Ap4HdlrAtom.cpp
index e31b3da..ea3d80a 100644
--- a/Source/C++/Core/Ap4HdlrAtom.cpp
+++ b/Source/C++/Core/Ap4HdlrAtom.cpp
@@ -144,8 +144,10 @@ AP4_HdlrAtom::WriteFields(AP4_ByteStream& stream)
     }
 
     // pad with zeros if necessary
-    AP4_Size padding = m_Size32 - (AP4_FULL_ATOM_HEADER_SIZE + 20 + name_size);
-    while (padding--) stream.WriteUI08(0);
+    if (m_Size32 > AP4_FULL_ATOM_HEADER_SIZE + 20 + name_size) {
+        AP4_Size padding = m_Size32 - (AP4_FULL_ATOM_HEADER_SIZE + 20 + name_size);
+        while (padding--) stream.WriteUI08(0);
+    }
 
     return AP4_SUCCESS;
 }
-- 
2.35.1

