--- a/src/dlfcn.c
+++ b/src/dlfcn.c
@@ -24,6 +24,7 @@
  * THE SOFTWARE.
  */
 
+#define _CRT_SECURE_NO_WARNINGS
 #ifdef _DEBUG
 #define _CRTDBG_MAP_ALLOC
 #include <stdlib.h>
@@ -311,6 +312,7 @@
 /* Load Psapi.dll at runtime, this avoids linking caveat */
 static BOOL MyEnumProcessModules( HANDLE hProcess, HMODULE *lphModule, DWORD cb, LPDWORD lpcbNeeded )
 {
+#if !defined(WINAPI_FAMILY) || (WINAPI_FAMILY != WINAPI_FAMILY_APP)
     static BOOL (WINAPI *EnumProcessModulesPtr)(HANDLE, HMODULE *, DWORD, LPDWORD) = NULL;
     static BOOL failed = FALSE;
     UINT uMode;
@@ -349,21 +351,27 @@
     }
 
     return EnumProcessModulesPtr( hProcess, lphModule, cb, lpcbNeeded );
+#else
+    return 0;
+#endif
 }
 
 DLFCN_EXPORT
 void *dlopen( const char *file, int mode )
 {
-    HMODULE hModule;
-    UINT uMode;
+    HMODULE hModule = NULL;
+    UINT uMode = 0;
 
     error_occurred = FALSE;
 
     /* Do not let Windows display the critical-error-handler message box */
+#if !defined(WINAPI_FAMILY) || (WINAPI_FAMILY != WINAPI_FAMILY_APP)
     uMode = MySetErrorMode( SEM_FAILCRITICALERRORS );
+#endif
 
     if( file == NULL )
     {
+#if !defined(WINAPI_FAMILY) || (WINAPI_FAMILY != WINAPI_FAMILY_APP) // what is replacement of GMH on UWP?
         /* POSIX says that if the value of file is NULL, a handle on a global
          * symbol object must be provided. That object must be able to access
          * all symbols from the original program file, and any objects loaded
@@ -378,6 +386,7 @@
 
         if( !hModule )
             save_err_str( "(null)", GetLastError( ) );
+#endif
     }
     else
     {
@@ -415,6 +424,34 @@
              * to UNIX's search paths (start with system folders instead of current
              * folder).
              */
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
+            int result = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, lpFileName, -1, NULL, 0);
+            if (result == 0)
+            {
+                save_err_str(lpFileName, GetLastError());
+                return NULL;
+            }
+
+            wchar_t* newStr = (wchar_t*)malloc(result*sizeof(wchar_t));
+            if (!newStr)
+            {
+                save_err_str(lpFileName, ERROR_NOT_ENOUGH_MEMORY);
+                return NULL;
+            }
+            result = MultiByteToWideChar( CP_UTF8, MB_ERR_INVALID_CHARS, lpFileName, -1, newStr, result );
+            if (result == 0)
+            {
+                save_err_str(lpFileName, GetLastError());
+                free( newStr );
+                return NULL;
+            }
+
+            hModule = LoadPackagedLibrary( newStr, 0 );
+            if (!hModule)
+                save_err_str(lpFileName, GetLastError());
+            free( newStr );
+            dwProcModsAfter = 0;
+#else // WINAPI_PARTITION_DESKTOP
             hModule = LoadLibraryExA( lpFileName, NULL, LOAD_WITH_ALTERED_SEARCH_PATH );
 
             if( !hModule )
@@ -449,11 +486,14 @@
                     local_rem( hModule );
                 }
             }
+#endif
         }
     }
 
     /* Return to previous state of the error-mode bit flags. */
+#if !defined(WINAPI_FAMILY) || (WINAPI_FAMILY != WINAPI_FAMILY_APP)
     MySetErrorMode( uMode );
+#endif
 
     return (void *) hModule;
 }
@@ -488,15 +528,16 @@
 {
     FARPROC symbol;
     HMODULE hCaller;
-    HMODULE hModule;
+    HMODULE hModule = 0;
     DWORD dwMessageId;
 
     error_occurred = FALSE;
 
     symbol = NULL;
+    dwMessageId = 0;
     hCaller = NULL;
+#if !defined(WINAPI_FAMILY) || (WINAPI_FAMILY != WINAPI_FAMILY_APP) // what is replacement of GMH on UWP?
     hModule = GetModuleHandle( NULL );
-    dwMessageId = 0;
 
     if( handle == RTLD_DEFAULT )
     {
@@ -525,6 +566,7 @@
             goto end;
         }
     }
+#endif
 
     if( handle != RTLD_NEXT )
     {
@@ -537,7 +579,7 @@
     /* If the handle for the original program file is passed, also search
      * in all globally loaded objects.
      */
-
+#if !defined(WINAPI_FAMILY) || (WINAPI_FAMILY != WINAPI_FAMILY_APP)
     if( hModule == handle || handle == RTLD_NEXT )
     {
         HANDLE hCurrentProc;
@@ -588,6 +630,7 @@
             }
         }
     }
+#endif
 
 end:
     if( symbol == NULL )
